### ISSUE TITLE
Review usages of ref vs in

### ISSUE THREAD

**Issue by RossNordby** [bepu/bepuphysics2#27]
If there are no compatibility blockers with [in parameters](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.2/readonly-ref.md), there are a whole bunch of places where they would be useful for reducing syntax noise. Consider reviewing most of the user-facing APIs for opportunities.

**Comment by RossNordby** [comment]
Some steps forward on this front have been made. I haven't found any cases where in parameters cause unexpectedly bad codegen, so I refactored some frequently used external APIs and the math libraries in fc8de883939112ced57c03f81f88eaf113149b27.

Overall performance change was within noise (1% better), but I have observed a few isolated cases where the in parameters actually did result in marginally better code quality for some reason. I'll take it.

(Notably, care must be taken around instance functions on in parameters. Unless the JIT is able to recognize that the function doesn't mutate the instance, it will trigger a protective copy. Unfortunately, it seems to be overprotective in many cases, so ref parameters and hoop-jumping are sometimes required.)
